freqtrade/optimize/optimize_reports.py

```diff
@@ -45,29 +45,41 @@ def store_backtest_stats(
     file_dump_json(latest_filename, {'latest_backtest': str(filename.name)})
 
 
-def store_backtest_signal_candles(
-        recordfilename: Path, candles: Dict[str, Dict], dtappendix: str) -> Path:
+def _store_backtest_analysis_data(
+        recordfilename: Path, data: Dict[str, Dict],
+        dtappendix: str, name: str) -> Path:
     """
-    Stores backtest trade signal candles
+    Stores backtest trade candles for analysis
     :param recordfilename: Path object, which can either be a filename or a directory.
         Filenames will be appended with a timestamp right before the suffix
-        while for directories, <directory>/backtest-result-<datetime>_signals.pkl will be used
+        while for directories, <directory>/backtest-result-<datetime>_<name>.pkl will be used
         as filename
-    :param stats: Dict containing the backtesting signal candles
+    :param candles: Dict containing the backtesting data for analysis
     :param dtappendix: Datetime to use for the filename
+    :param name: Name to use for the file, e.g. signals, rejected
     """
     if recordfilename.is_dir():
-        filename = (recordfilename / f'backtest-result-{dtappendix}_signals.pkl')
+        filename = (recordfilename / f'backtest-result-{dtappendix}_{name}.pkl')
     else:
         filename = Path.joinpath(
-            recordfilename.parent, f'{recordfilename.stem}-{dtappendix}_signals.pkl'
+            recordfilename.parent, f'{recordfilename.stem}-{dtappendix}_{name}.pkl'
         )
 
-    file_dump_joblib(filename, candles)
+    file_dump_joblib(filename, data)
 
     return filename
 
 
+def store_backtest_signal_candles(
+        recordfilename: Path, candles: Dict[str, Dict], dtappendix: str) -> Path:
+    return _store_backtest_analysis_data(Path(recordfilename), candles, dtappendix, "signals")
+
+
+def store_backtest_rejected_signals(
+        recordfilename: Path, trades: Dict[str, Dict], dtappendix: str) -> Path:
+    return _store_backtest_analysis_data(Path(recordfilename), trades, dtappendix, "rejected")
```

Comment:
Considering how these 2 functions are used - it might make more sense to have just 1 function - which calls the same underlying function twice with diff. arguments.

That'll keep the part in backtesting file cleaner - as it's only one call - abstracting away the logic as to WHAT you're doing with the files below.
it's not like there's an option to call only one or the other ...

Reply:
OK, sounds fair.

Reply:
Would you suggest using an enum for the type? e.g. `freqtrade.enums.BacktestingAnalysisOutputType.SIGNAL_CANDLES`

Reply:
what for? (maybe i'm missing some context ...) 

it's not really optional (at least not right now)
if you call one - you'll also call the other.

My point above was to use one method - and simply call `_store_backtest_analysis_data` twice from within the same method - so i don't really see a usecase for an enum?

Reply:
Ah sorry, I was overthinking it. Understood 👍